src/v/kafka/protocol/batch_consumer.h

```diff
@@ -48,6 +55,10 @@ class kafka_batch_serializer {
         if (unlikely(record_count_ == 0)) {
             _base_offset = batch.base_offset();
         }
+        if (
+          batch.header().attrs.is_transactional() && !_first_tx_batch_offset) {
```

Comment:
maybe it would make sense to flip the expressions order this way after the `_first_tx_batch_offset` the other statement wouldn't have to be evaluated also we can wrap it with `unlikely`